hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -104,12 +103,12 @@ public void initialize(String inputName, Map<String, String> properties) {
     }
 
     if (properties.containsKey(CatalogProperties.URI)) {
-      this.conf.set(HiveConf.ConfVars.METASTOREURIS.varname, properties.get(CatalogProperties.URI));
+      this.conf.set("hive.metastore.uris", properties.get(CatalogProperties.URI));
```

Comment:
On the other PR (https://github.com/apache/iceberg/pull/12681#discussion_r2030387861) @manuzhang mentioned that we might want to create constants for the keys.
While I agree with you that we don't want to expose the constants for the users (they just should use the Hive ones), we might want to create `@VisibleForTesting` package private constants to get rid of the strings all over the code.

Reply:
Peter, package private constants will not be accessible in the Flink and Spark modules that also need to access the Hive conf (mostly in test code, but also in `SparkSessionCatalog`). See https://github.com/apache/iceberg/pull/12693.
I don't think this is worthwhile.